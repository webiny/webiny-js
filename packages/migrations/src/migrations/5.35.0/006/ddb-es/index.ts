import { Table } from "dynamodb-toolbox";
import { Client } from "@elastic/elasticsearch";
import {
    DataMigration,
    DataMigrationContext,
    ElasticsearchClientSymbol,
    ElasticsearchDynamoTableSymbol,
    getChildLogger,
    PrimaryDynamoTableSymbol
} from "@webiny/data-migration";
import { makeInjectable, inject } from "@webiny/ioc";
import { AcoRecords_5_35_0_006_PageData } from "./PageDataMigration";
export * from "../types";

export class AcoRecords_5_35_0_006 implements DataMigration {
    private migrations: DataMigration[];

    constructor(table: Table, esTable: Table, elasticsearchClient: Client) {
        this.migrations = [new AcoRecords_5_35_0_006_PageData(table, esTable, elasticsearchClient)];
    }

    getId(): string {
        return "5.35.0-006";
    }

    getDescription(): string {
        return "ACO search record migration";
    }

    async shouldExecute(context: DataMigrationContext): Promise<boolean> {
        for (const migration of this.migrations) {
            const childLogger = getChildLogger(context.logger, migration);
            const childContext = { ...context, logger: childLogger };
            if (await migration.shouldExecute(childContext)) {
                return true;
            }
        }
        return false;
    }

    async execute(context: DataMigrationContext): Promise<void> {
        for (const migration of this.migrations) {
            const childLogger = getChildLogger(context.logger, migration);
            const childContext = { ...context, logger: childLogger };
            if (await migration.shouldExecute(childContext)) {
                await migration.execute(childContext);
            }
        }
    }
}

makeInjectable(AcoRecords_5_35_0_006, [
    inject(PrimaryDynamoTableSymbol),
    inject(ElasticsearchDynamoTableSymbol),
    inject(ElasticsearchClientSymbol)
]);
