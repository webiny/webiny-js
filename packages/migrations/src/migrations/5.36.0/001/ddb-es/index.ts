import { Table } from "dynamodb-toolbox";
import { Client } from "@elastic/elasticsearch";
import {
    DataMigration,
    DataMigrationContext,
    ElasticsearchClientSymbol,
    ElasticsearchDynamoTableSymbol,
    getChildLogger,
    PrimaryDynamoTableSymbol
} from "@webiny/data-migration";
import { makeInjectable, inject } from "@webiny/ioc";
import { AcoRecords_5_36_0_001_FileData } from "./FileDataMigration";
export * from "../types";

export class AcoRecords_5_36_0_001 implements DataMigration {
    private migrations: DataMigration[];

    constructor(table: Table, esTable: Table, elasticsearchClient: Client) {
        this.migrations = [new AcoRecords_5_36_0_001_FileData(table, esTable, elasticsearchClient)];
    }

    getId(): string {
        return "5.36.0-001";
    }

    getDescription(): string {
        return "Migrate FmFile Data -> Create ACO Search Records";
    }

    async shouldExecute(context: DataMigrationContext): Promise<boolean> {
        for (const migration of this.migrations) {
            const childLogger = getChildLogger(context.logger, migration);
            const childContext = { ...context, logger: childLogger };
            if (await migration.shouldExecute(childContext)) {
                return true;
            }
        }
        return false;
    }

    async execute(context: DataMigrationContext): Promise<void> {
        for (const migration of this.migrations) {
            const childLogger = getChildLogger(context.logger, migration);
            const childContext = { ...context, logger: childLogger };
            if (await migration.shouldExecute(childContext)) {
                await migration.execute(childContext);
            }
        }
    }
}

makeInjectable(AcoRecords_5_36_0_001, [
    inject(PrimaryDynamoTableSymbol),
    inject(ElasticsearchDynamoTableSymbol),
    inject(ElasticsearchClientSymbol)
]);
