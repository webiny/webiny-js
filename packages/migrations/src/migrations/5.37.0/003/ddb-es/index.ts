import { Table } from "dynamodb-toolbox";
import { Client } from "@elastic/elasticsearch";
import {
    DataMigration,
    DataMigrationContext,
    ElasticsearchClientSymbol,
    ElasticsearchDynamoTableSymbol,
    getChildLogger,
    PrimaryDynamoTableSymbol
} from "@webiny/data-migration";
import { inject, makeInjectable } from "@webiny/ioc";
import { AcoRecords_5_37_0_003_PageData } from "./PageDataMigration";

export * from "../types";

export class AcoRecords_5_37_0_003 implements DataMigration {
    private readonly migrations: DataMigration[];

    constructor(table: Table, esTable: Table, elasticsearchClient: Client) {
        this.migrations = [new AcoRecords_5_37_0_003_PageData(table, esTable, elasticsearchClient)];
    }

    getId(): string {
        return "5.37.0-003";
    }

    getDescription(): string {
        return "Page Builder Pages search record migration";
    }

    async shouldExecute(context: DataMigrationContext): Promise<boolean> {
        for (const migration of this.migrations) {
            const childLogger = getChildLogger(context.logger, migration);
            const childContext = { ...context, logger: childLogger };
            if (await migration.shouldExecute(childContext)) {
                return true;
            }
        }
        return false;
    }

    async execute(context: DataMigrationContext): Promise<void> {
        for (const migration of this.migrations) {
            const childLogger = getChildLogger(context.logger, migration);
            const childContext = { ...context, logger: childLogger };
            if (await migration.shouldExecute(childContext)) {
                await migration.execute(childContext);
            }
        }
    }
}

makeInjectable(AcoRecords_5_37_0_003, [
    inject(PrimaryDynamoTableSymbol),
    inject(ElasticsearchDynamoTableSymbol),
    inject(ElasticsearchClientSymbol)
]);
