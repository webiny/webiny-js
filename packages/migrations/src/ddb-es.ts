// 5.35.0
import { FileManager_5_35_0_001 } from "./migrations/5.35.0/001/ddb-es";
import { PageBuilder_5_35_0_002 } from "~/migrations/5.35.0/002";
import { AdminUsers_5_35_0_003 } from "~/migrations/5.35.0/003";
import { Tenancy_5_35_0_004 } from "~/migrations/5.35.0/004";
import { CmsModels_5_35_0_005 } from "~/migrations/5.35.0/005";
import { AcoRecords_5_35_0_006 } from "~/migrations/5.35.0/006/ddb-es";
// 5.36.0
import { AcoRecords_5_36_0_001 } from "~/migrations/5.36.0/001/ddb-es";
// 5.37.0
import { TenantLinkRecords_5_37_0_001 } from "~/migrations/5.37.0/001";
import { CmsEntriesRootFolder_5_37_0_002 } from "~/migrations/5.37.0/002/ddb-es";
import { AcoFolders_5_37_0_003 } from "~/migrations/5.37.0/003/ddb-es";
import { AcoRecords_5_37_0_004 } from "~/migrations/5.37.0/004/ddb-es";
import { FileManager_5_37_0_005 } from "~/migrations/5.37.0/005/ddb-es";
// 5.38.0
import { MultiStepForms_5_38_0_001 } from "~/migrations/5.38.0/001/ddb-es";
import { MultiStepForms_5_38_0_002 } from "~/migrations/5.38.0/002/ddb-es";
// Page Blocks storage is the same for both DDB abd DDB-ES projects.
import { PageBlocks_5_38_0_003 } from "~/migrations/5.38.0/003/ddb";
// 5.39.0
import { CmsEntriesInitNewMetaFields_5_39_0_001 } from "~/migrations/5.39.0/001/ddb-es";
import { FileManager_5_39_0_005 } from "~/migrations/5.39.0/005/ddb-es";

export const migrations = () => {
    return [
        FileManager_5_35_0_001,
        PageBuilder_5_35_0_002,
        AdminUsers_5_35_0_003,
        Tenancy_5_35_0_004,
        CmsModels_5_35_0_005,
        AcoRecords_5_35_0_006,
        AcoRecords_5_36_0_001,
        TenantLinkRecords_5_37_0_001,
        CmsEntriesRootFolder_5_37_0_002,
        AcoFolders_5_37_0_003,
        AcoRecords_5_37_0_004,
        FileManager_5_37_0_005,
        MultiStepForms_5_38_0_001,
        MultiStepForms_5_38_0_002,
        PageBlocks_5_38_0_003,
        CmsEntriesInitNewMetaFields_5_39_0_001,
        FileManager_5_39_0_005
    ];
};
