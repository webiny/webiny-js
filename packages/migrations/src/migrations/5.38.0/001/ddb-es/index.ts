import { Table } from "dynamodb-toolbox";
import {
    DataMigration,
    DataMigrationContext,
    ElasticsearchClientSymbol,
    PrimaryDynamoTableSymbol
} from "@webiny/data-migration";
import { createFormEntity } from "../entities/createFormEntity";
import {
    batchReadAll,
    BatchReadItem,
    batchWriteAll,
    BatchWriteItem,
    esGetIndexExist,
    esGetIndexName,
    esQueryAll,
    queryAll
} from "~/utils";
import { FbForm } from "../types";
import { inject, makeInjectable } from "@webiny/ioc";
import { I18NLocale, ListLocalesParams, Tenant } from "~/migrations/5.37.0/003/types";
import { createLocaleEntity } from "../entities/createLocaleEntity";
import { createTenantEntity } from "../entities/createTenantEntity";
import { Client } from "@elastic/elasticsearch";
import { executeWithRetry } from "@webiny/utils";

export class MultiStepForms_5_38_0_001 implements DataMigration {
    private readonly formEntity: ReturnType<typeof createFormEntity>;
    private readonly localeEntity: ReturnType<typeof createLocaleEntity>;
    private readonly tenantEntity: ReturnType<typeof createTenantEntity>;
    private readonly elasticsearchClient: Client;

    constructor(table: Table, elasticsearchClient: Client) {
        this.formEntity = createFormEntity(table);
        this.localeEntity = createLocaleEntity(table);
        this.tenantEntity = createTenantEntity(table);
        this.elasticsearchClient = elasticsearchClient;
    }

    getId() {
        return "5.38.0-001";
    }

    getDescription() {
        return "Convert forms tu multi-step forms.";
    }

    async shouldExecute({ logger }: DataMigrationContext): Promise<boolean> {
        const tenants = await this.listTenants();
        if (tenants.length === 0) {
            logger.info(`No tenants found in the system; skipping migration.`);
            return false;
        }

        for (const tenant of tenants) {
            const locales = await this.listLocales({ tenant });
            if (locales.length === 0) {
                logger.info(`No locales found in tenant "${tenant.data.id}".`);
                continue;
            }

            for (const locale of locales) {
                const indexNameParams = {
                    tenant: tenant.data.id,
                    locale: locale.code,
                    type: "form-builder",
                    isHeadlessCmsModel: false
                };

                const indexExists = await esGetIndexExist({
                    elasticsearchClient: this.elasticsearchClient,
                    ...indexNameParams
                });

                if (!indexExists) {
                    logger.info(
                        `No Elasticsearch index found for folders in tenant "${tenant.data.id}" and locale "${locale.code}"; skipping.`
                    );
                    continue;
                }

                const records = await esQueryAll<FbForm>({
                    elasticsearchClient: this.elasticsearchClient,
                    index: esGetIndexName(indexNameParams),
                    body: {
                        size: 1000
                    }
                });

                const batchGetItems: BatchReadItem[] = [];
                for (let i = 0; i < records.length; i++) {
                    const record = records[i];
                    batchGetItems.push(
                        this.formEntity.getBatch({
                            // T#root#L#en-US#FB#F#653a578347a0da00088b9f2f
                            PK: `T#${tenant.data.id}#L#${locale.code}#FB#F#${record.formId}`,
                            SK: "L"
                        })
                    );
                }

                // Get DynamoDB records for all the forms retrieved from Elasticsearch.
                const ddbRecords = await batchReadAll<FbForm>({
                    table: this.formEntity.table,
                    items: batchGetItems
                });

                for (let i = 0; i < ddbRecords.length; i++) {
                    const current = ddbRecords[i];
                    if (!current.steps) {
                        return true;
                    }
                }
            }
        }

        return false;
    }

    async execute({ logger }: DataMigrationContext): Promise<void> {
        const tenants = await this.listTenants();

        for (const tenant of tenants) {
            const locales = await this.listLocales({ tenant });
            if (locales.length === 0) {
                continue;
            }

            for (const locale of locales) {
                const indexNameParams = {
                    tenant: tenant.data.id,
                    locale: locale.code,
                    type: "form-builder",
                    isHeadlessCmsModel: false
                };

                const indexExists = await esGetIndexExist({
                    elasticsearchClient: this.elasticsearchClient,
                    ...indexNameParams
                });

                if (!indexExists) {
                    continue;
                }

                const records = await esQueryAll<FbForm>({
                    elasticsearchClient: this.elasticsearchClient,
                    index: esGetIndexName(indexNameParams),
                    body: {
                        size: 1000
                    }
                });

                for (let i = 0; i < records.length; i++) {
                    const record = records[i];
                    const tenantId = tenant.data.id;
                    const localeCode = locale.code;
                    const formId = record.formId;

                    const formRecords = await queryAll<FbForm>({
                        entity: this.formEntity,
                        partitionKey: `T#${tenantId}#L#${localeCode}#FB#F#${formId}`,
                        options: {
                            gte: " "
                        }
                    });

                    const items: BatchWriteItem[] = [];
                    for (const current of formRecords) {
                        const isFbForm = current.TYPE?.startsWith("fb.form");
                        if (!isFbForm) {
                            continue;
                        }

                        if (current.steps) {
                            continue;
                        }

                        // If no steps are defined, we need to create a single step.
                        current.steps = [];

                        if (Array.isArray(current.layout)) {
                            // If layout is an array, we need to create a single step with all the fields.
                            current.steps.push({ title: "Step 1", layout: current.layout });
                            delete current.layout;
                        }

                        items.push(this.formEntity.putBatch(current));
                    }

                    if (items.length === 0) {
                        return;
                    }

                    const execute = () => {
                        return batchWriteAll({ table: this.formEntity.table, items });
                    };

                    logger.trace("Storing the DynamoDB records...");
                    await executeWithRetry(execute, {
                        onFailedAttempt: error => {
                            logger.error(`"batchWriteAll" attempt #${error.attemptNumber} failed.`);
                            logger.error(error.message);
                        }
                    });
                }
            }
        }

        logger.info("Updated all the forms.");
    }

    private async listTenants(): Promise<Tenant[]> {
        return await queryAll<Tenant>({
            entity: this.tenantEntity,
            partitionKey: "TENANTS",
            options: {
                index: "GSI1",
                gte: " "
            }
        });
    }

    private async listLocales({ tenant }: ListLocalesParams): Promise<I18NLocale[]> {
        return await queryAll<I18NLocale>({
            entity: this.localeEntity,
            partitionKey: `T#${tenant.data.id}#I18N#L`,
            options: {
                gte: " "
            }
        });
    }
}

makeInjectable(MultiStepForms_5_38_0_001, [
    inject(PrimaryDynamoTableSymbol),
    inject(ElasticsearchClientSymbol)
]);
