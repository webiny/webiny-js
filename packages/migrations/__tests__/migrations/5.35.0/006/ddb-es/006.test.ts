import { createElasticsearchClient } from "@webiny/project-utils/testing/elasticsearch/client";

import {
    assertNotError,
    createDdbEsMigrationHandler,
    delay,
    getDynamoToEsTable,
    getPrimaryDynamoDbTable,
    groupMigrations,
    insertDynamoDbTestData as insertTestData,
    logTestNameBeforeEachTest,
    scanTable
} from "~tests/utils";

import { AcoRecords_5_35_0_006 } from "~/migrations/5.35.0/006/ddb-es";

import {
    createTenantsData,
    createLocalesData,
    createDdbEsPagesData,
    createDdbPagesData
} from "./006.data";

jest.retryTimes(0);
jest.setTimeout(900000);

describe("5.35.0-006", () => {
    const ddbTable = getPrimaryDynamoDbTable();
    const ddbToEsTable = getDynamoToEsTable();
    const elasticsearchClient = createElasticsearchClient();

    beforeAll(() => {
        process.env.ELASTIC_SEARCH_INDEX_PREFIX =
            new Date().toISOString().replace(/\.|\:/g, "-").toLowerCase() + "-";
    });

    logTestNameBeforeEachTest();

    it("should not run if no tenant found", async () => {
        const handler = createDdbEsMigrationHandler({
            primaryTable: ddbTable,
            dynamoToEsTable: ddbToEsTable,
            elasticsearchClient,
            migrations: [AcoRecords_5_35_0_006]
        });

        const { data, error } = await handler();

        assertNotError(error);
        const grouped = groupMigrations(data.migrations);

        expect(grouped.executed.length).toBe(0);
        expect(grouped.skipped.length).toBe(1);
        expect(grouped.notApplicable.length).toBe(0);
    });

    it("should not run if no locale found", async () => {
        await insertTestData(ddbTable, [...createTenantsData()]);

        const handler = createDdbEsMigrationHandler({
            primaryTable: ddbTable,
            dynamoToEsTable: ddbToEsTable,
            elasticsearchClient,
            migrations: [AcoRecords_5_35_0_006]
        });

        const { data, error } = await handler();

        assertNotError(error);
        const grouped = groupMigrations(data.migrations);

        expect(grouped.executed.length).toBe(0);
        expect(grouped.skipped.length).toBe(1);
        expect(grouped.notApplicable.length).toBe(0);
    });

    it("should not run if no pages found", async () => {
        await insertTestData(ddbTable, [...createTenantsData(), ...createLocalesData()]);

        const handler = createDdbEsMigrationHandler({
            primaryTable: ddbTable,
            dynamoToEsTable: ddbToEsTable,
            elasticsearchClient,
            migrations: [AcoRecords_5_35_0_006]
        });

        const { data, error } = await handler();

        assertNotError(error);
        const grouped = groupMigrations(data.migrations);

        expect(grouped.executed.length).toBe(0);
        expect(grouped.skipped.length).toBe(1);
        expect(grouped.notApplicable.length).toBe(0);
    });

    it("should execute migration", async () => {
        await insertTestData(ddbTable, [
            ...createTenantsData(),
            ...createLocalesData(),
            ...createDdbPagesData()
        ]);

        await insertTestData(ddbToEsTable, [...createDdbEsPagesData()]);

        const handler = createDdbEsMigrationHandler({
            primaryTable: ddbTable,
            dynamoToEsTable: ddbToEsTable,
            elasticsearchClient,
            migrations: [AcoRecords_5_35_0_006]
        });

        const { data, error } = await handler();

        assertNotError(error);
        const grouped = groupMigrations(data.migrations);

        expect(grouped.executed.length).toBe(1);
        expect(grouped.skipped.length).toBe(0);
        expect(grouped.notApplicable.length).toBe(0);

        const ddbSearchRecords = await scanTable(ddbTable, {
            entity: "CmsEntries",
            filters: [
                {
                    attr: "modelId",
                    eq: "acoSearchRecord"
                }
            ]
        });

        expect(ddbSearchRecords.length).toBe(4);

        // Test result with snapshots - for some fields we need to use property matchers
        ddbSearchRecords.forEach(record => {
            expect(record).toMatchSnapshot({
                created: expect.any(String),
                modified: expect.any(String),
                webinyVersion: expect.any(String)
            });
        });

        const ddbEsSearchRecords = await scanTable(ddbToEsTable, {
            entity: "CmsEntries",
            filters: [
                {
                    attr: "index",
                    contains: "acosearchrecord"
                }
            ]
        });

        expect(ddbEsSearchRecords.length).toBe(2);

        ddbEsSearchRecords.forEach(record => {
            expect(record).toMatchSnapshot({
                created: expect.any(String),
                modified: expect.any(String)
            });
        });
    });

    it("should not run migration if data is already in the expected shape", async () => {
        await insertTestData(ddbTable, [
            ...createTenantsData(),
            ...createLocalesData(),
            ...createDdbPagesData()
        ]);

        await insertTestData(ddbToEsTable, [...createDdbEsPagesData()]);

        await delay(3000);

        const handler = createDdbEsMigrationHandler({
            primaryTable: ddbTable,
            dynamoToEsTable: ddbToEsTable,
            elasticsearchClient,
            migrations: [AcoRecords_5_35_0_006]
        });

        // Should run the migration
        process.stdout.write("[First run]\n");
        const firstRun = await handler();
        assertNotError(firstRun.error);
        const firstData = groupMigrations(firstRun.data.migrations);
        expect(firstData.executed.length).toBe(1);

        // Should skip the migration
        process.stdout.write("[Second run]\n");
        const secondRun = await handler();
        assertNotError(secondRun.error);
        const secondData = groupMigrations(secondRun.data.migrations);
        expect(secondData.executed.length).toBe(0);
        expect(secondData.skipped.length).toBe(1);
        expect(secondData.notApplicable.length).toBe(0);
    });
});
