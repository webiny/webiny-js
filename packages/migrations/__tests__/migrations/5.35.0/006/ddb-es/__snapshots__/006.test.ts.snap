// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`5.35.0-006 should execute migration 1`] = `
Object {
  "PK": "T#root#L#en-US#CMS#CME#642d3f644d7cdc000821b67e",
  "SK": "L",
  "TYPE": "cms.entry.l",
  "created": Any<String>,
  "createdBy": Object {
    "displayName": "Admin",
    "id": "admin",
    "type": "admin",
  },
  "createdOn": "2023-04-05T09:29:08.538Z",
  "entity": "CmsEntries",
  "entryId": "642d3f644d7cdc000821b67e",
  "id": "642d3f644d7cdc000821b67e#0001",
  "locale": "en-US",
  "locked": false,
  "meta": Object {},
  "modelId": "acoSearchRecord",
  "modified": Any<String>,
  "modifiedBy": Object {
    "displayName": "Admin",
    "id": "admin",
    "type": "admin",
  },
  "ownedBy": Object {
    "displayName": "Admin",
    "id": "admin",
    "type": "admin",
  },
  "savedOn": "2023-04-05T09:32:01.638Z",
  "status": "draft",
  "tenant": "root",
  "values": Object {
    "content": "Page 1 Page 1 Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse varius enim in eros elementum tristique. Duis cursus, mi quis viverra ornare, eros dolor interdum nulla, ut commodo diam libero vitae erat.",
    "data": Object {
      "createdBy": Object {
        "displayName": "Admin",
        "id": "admin",
        "type": "admin",
      },
      "createdOn": "2023-04-05T09:29:08.538Z",
      "id": "642d3f644d7cdc000821b67e#0001",
      "locked": true,
      "path": "/untitled-lg3hm56i",
      "pid": "642d3f644d7cdc000821b67e",
      "savedOn": "2023-04-05T09:32:01.638Z",
      "status": "published",
      "title": "Page 1",
      "version": 1,
    },
    "location": Object {
      "folderId": "ROOT",
    },
    "title": "Page 1",
    "type": "PbPage",
  },
  "version": 1,
  "webinyVersion": Any<String>,
}
`;

exports[`5.35.0-006 should execute migration 2`] = `
Object {
  "PK": "T#root#L#en-US#CMS#CME#642d3f644d7cdc000821b67e",
  "SK": "REV#0001",
  "TYPE": "cms.entry",
  "created": Any<String>,
  "createdBy": Object {
    "displayName": "Admin",
    "id": "admin",
    "type": "admin",
  },
  "createdOn": "2023-04-05T09:29:08.538Z",
  "entity": "CmsEntries",
  "entryId": "642d3f644d7cdc000821b67e",
  "id": "642d3f644d7cdc000821b67e#0001",
  "locale": "en-US",
  "locked": false,
  "meta": Object {},
  "modelId": "acoSearchRecord",
  "modified": Any<String>,
  "modifiedBy": Object {
    "displayName": "Admin",
    "id": "admin",
    "type": "admin",
  },
  "ownedBy": Object {
    "displayName": "Admin",
    "id": "admin",
    "type": "admin",
  },
  "savedOn": "2023-04-05T09:32:01.638Z",
  "status": "draft",
  "tenant": "root",
  "values": Object {
    "content": "Page 1 Page 1 Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse varius enim in eros elementum tristique. Duis cursus, mi quis viverra ornare, eros dolor interdum nulla, ut commodo diam libero vitae erat.",
    "data": Object {
      "createdBy": Object {
        "displayName": "Admin",
        "id": "admin",
        "type": "admin",
      },
      "createdOn": "2023-04-05T09:29:08.538Z",
      "id": "642d3f644d7cdc000821b67e#0001",
      "locked": true,
      "path": "/untitled-lg3hm56i",
      "pid": "642d3f644d7cdc000821b67e",
      "savedOn": "2023-04-05T09:32:01.638Z",
      "status": "published",
      "title": "Page 1",
      "version": 1,
    },
    "location": Object {
      "folderId": "ROOT",
    },
    "title": "Page 1",
    "type": "PbPage",
  },
  "version": 1,
  "webinyVersion": Any<String>,
}
`;

exports[`5.35.0-006 should execute migration 3`] = `
Object {
  "PK": "T#root#L#en-US#CMS#CME#642d401c4d7cdc000821b681",
  "SK": "L",
  "TYPE": "cms.entry.l",
  "created": Any<String>,
  "createdBy": Object {
    "displayName": "Admin",
    "id": "admin",
    "type": "admin",
  },
  "createdOn": "2023-04-05T09:32:12.436Z",
  "entity": "CmsEntries",
  "entryId": "642d401c4d7cdc000821b681",
  "id": "642d401c4d7cdc000821b681#0001",
  "locale": "en-US",
  "locked": false,
  "meta": Object {},
  "modelId": "acoSearchRecord",
  "modified": Any<String>,
  "modifiedBy": Object {
    "displayName": "Admin",
    "id": "admin",
    "type": "admin",
  },
  "ownedBy": Object {
    "displayName": "Admin",
    "id": "admin",
    "type": "admin",
  },
  "savedOn": "2023-04-05T09:37:04.998Z",
  "status": "draft",
  "tenant": "root",
  "values": Object {
    "content": "Page 2 Page 2 Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ut sem vitae risus tristique posuere.",
    "data": Object {
      "createdBy": Object {
        "displayName": "Admin",
        "id": "admin",
        "type": "admin",
      },
      "createdOn": "2023-04-05T09:32:12.436Z",
      "id": "642d401c4d7cdc000821b681#0001",
      "locked": false,
      "path": "/untitled-lg3hq32s",
      "pid": "642d401c4d7cdc000821b681",
      "savedOn": "2023-04-05T09:37:04.998Z",
      "status": "draft",
      "title": "Page 2",
      "version": 1,
    },
    "location": Object {
      "folderId": "ROOT",
    },
    "title": "Page 2",
    "type": "PbPage",
  },
  "version": 1,
  "webinyVersion": Any<String>,
}
`;

exports[`5.35.0-006 should execute migration 4`] = `
Object {
  "PK": "T#root#L#en-US#CMS#CME#642d401c4d7cdc000821b681",
  "SK": "REV#0001",
  "TYPE": "cms.entry",
  "created": Any<String>,
  "createdBy": Object {
    "displayName": "Admin",
    "id": "admin",
    "type": "admin",
  },
  "createdOn": "2023-04-05T09:32:12.436Z",
  "entity": "CmsEntries",
  "entryId": "642d401c4d7cdc000821b681",
  "id": "642d401c4d7cdc000821b681#0001",
  "locale": "en-US",
  "locked": false,
  "meta": Object {},
  "modelId": "acoSearchRecord",
  "modified": Any<String>,
  "modifiedBy": Object {
    "displayName": "Admin",
    "id": "admin",
    "type": "admin",
  },
  "ownedBy": Object {
    "displayName": "Admin",
    "id": "admin",
    "type": "admin",
  },
  "savedOn": "2023-04-05T09:37:04.998Z",
  "status": "draft",
  "tenant": "root",
  "values": Object {
    "content": "Page 2 Page 2 Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ut sem vitae risus tristique posuere.",
    "data": Object {
      "createdBy": Object {
        "displayName": "Admin",
        "id": "admin",
        "type": "admin",
      },
      "createdOn": "2023-04-05T09:32:12.436Z",
      "id": "642d401c4d7cdc000821b681#0001",
      "locked": false,
      "path": "/untitled-lg3hq32s",
      "pid": "642d401c4d7cdc000821b681",
      "savedOn": "2023-04-05T09:37:04.998Z",
      "status": "draft",
      "title": "Page 2",
      "version": 1,
    },
    "location": Object {
      "folderId": "ROOT",
    },
    "title": "Page 2",
    "type": "PbPage",
  },
  "version": 1,
  "webinyVersion": Any<String>,
}
`;

exports[`5.35.0-006 should execute migration 5`] = `
Object {
  "PK": "T#root#L#en-US#CMS#CME#642d3f644d7cdc000821b67e",
  "SK": "L",
  "created": Any<String>,
  "data": Object {
    "compression": "gzip",
    "value": "H4sIAAAAAAAAE6VUTW/bMAz9K4J2dTzns2luG7bDgGEJmqxAdykYi0mIyZIrUQmCIP99tPMBbCi6AoEPFp/IJ/KR0kFX3qD9ZvREQ+nnCKHcPGDpg9GZ3mKI5J2edDMdYYtmKmvdK3r9TjHoFMNFcT/p9yZFNx/1x78kwPoSLIoPus7PuQCRgVMUwARYcUMJNqEAB837uvGcLWewRtlh4ja0MVVXgNI7RsdXSJ1/333ASlEdU6WMtz6oSKygQs6UxEQsGTkFBYZqiiW5tUJLnKt5ijU6QzGi2kKgFBU6EiqnMHgxLFZyoNByoMj0kjBXXxJFVaYQU8xUReqlsbck0gRQPjgImJ3CT7mQJB2McLhkLWQqsSRVicpeGYJKWVqKtzAwoMQB52fduFX6oFfeGgxtRx6m04U+HkWKgMBoPu8bB2qbZSpyjWonFS+mFFdb2P8QNQT91KISL1WFfUs5GvRMfzUaDMxdacqiKMa97nJ01zYAHbRyB++bVp0PfaXnvftJMc6H157/RqFegY2Yab9zNyS6wyW5/eNl7nSRyyfu9FbuH2TZDIyITCu64XQr9UZRgEOSShZPs6+yW1Yxb/XLrUQ+P5+p/oYD7B6vk22A4ZrB/5Ku3/R69V7cOAzv7Ou7Lvz1ftdpaSlu8N9n4zIdJ0lr4I04f0yurcp07Lq/qYYjkik//gFiJ98pjAQAAA==",
  },
  "entity": "CmsEntries",
  "index": "root-headless-cms-en-us-acosearchrecord",
  "modified": Any<String>,
}
`;

exports[`5.35.0-006 should execute migration 6`] = `
Object {
  "PK": "T#root#L#en-US#CMS#CME#642d401c4d7cdc000821b681",
  "SK": "L",
  "created": Any<String>,
  "data": Object {
    "compression": "gzip",
    "value": "H4sIAAAAAAAAE6VTTW8aMRD9K5Z7ha33owT21ko9VKoCSmik9hIZewCrXntjj0EI8d87u9mgNonSSMgH288z43kzb4688RrsN81rLpW/BRnU9gaUD5qP+A5CNN7xOh/xKHeg53TmhSjKsajG4tNSzOryqhZVNptNf5GD9UpaIBtw4x+3BESUmCIBOsg1diGlTUDAkeOh7SwXq4XcAL2gwd61u7KCAOUdgsMzxIbtuw/QMNPG1DDtrQ8sGmSyARwx8omgEDAFJrVpTVTGbRhYgxm7Tk6xhCyS+86gBBZMTJEhbWgeErDWxwQBsoEK9uSPfO2thtAX6WY+X/LTibILIBH0l0NnYPr66ca4jsgjsaerNrG18nBNCRL6uUfJn5iFQx9yUhW6Ermq9JXSSggxLfLVZJp3ocDJvgLB+656w6evtaGo8yKrysnQht9AodfSRhhxv3cXJLqHlXGHuycpcJHRInPzVu4f6NgRIHWZtbngd0t8I1UAQyImy5+Lr/Sqmpj19cssed7fD6H+hYPc353FpiXKcwb/S7p90+pVqV4ohnf29V0z+HLk/priZ8poJW7J8mNyPSU9tpty+1AWkSR++gOp5TpPHAQAAA==",
  },
  "entity": "CmsEntries",
  "index": "root-headless-cms-en-us-acosearchrecord",
  "modified": Any<String>,
}
`;
